dotCMS/src/main/java/com/dotcms/rest/api/v1/page/NavResource.java

```diff
@@ -91,13 +93,17 @@ protected NavResource(final WebResource webResource) {
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
     @Path("/{uri: .*}")
     public final Response loadJson(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
-            @PathParam("uri") final String uri, @QueryParam("depth") final String depth) {
+            @PathParam("uri") final String uri, @QueryParam("depth") final String depth, @QueryParam("languageId") final String languageId) {
 
         final InitDataObject auth = webResource.init(false, request, true);
         final User user = auth.getUser();
 
         try {
-            final int maxDepth = Integer.parseInt(depth);
+            final int maxDepth = (UtilMethods.isSet(depth)) ? Integer.parseInt(depth) :
```

Comment:
ConversionUtils.toInt(depth, defaultDepth);

Reply:
This does not apply, because the conversionUtils method is catching the NumberFormatException and returning the default value since this is a Resource if the user inputs a bad parameter, should return a 400 error.